docs/rules/no-commonjs.md

```diff
@@ -21,6 +21,24 @@ module.exports = { a: "b" }
 exports.c = "d"
 ```
 
+### Allow require
+
+If `allow-require` is provided as an option, `require` calls are valid:
+
+```js
+/*eslint no-commonjs: [2, "allow-require"]*/
+
+if (typeof window !== "undefined") {
+  require('that-ugly-thing');
+}
+```
+
+but `module.exports` is reported as usual.
+
+This is useful for conditional requires.
```

Comment:
Actually, I just looked though some [babel-plugin-dynamic-import-node test](https://github.com/airbnb/babel-plugin-dynamic-import-node/blob/master/test/fixtures/basic-import/expected.es2015.js), the new dynamic `import` is async and thus is not a drop-in replacement for `require`. So I guess adding a hint like this would suffice:
> This is useful for conditional requires. If you don't rely on synchronous module loading, check out [dynamic import](https://github.com/airbnb/babel-plugin-dynamic-import-node).